src/TileExpandable/index.js

```diff
@@ -0,0 +1,124 @@
+  flex-direction: column;
+  width: 100%;
+  background: ${({ theme }) => theme.orbit.paletteWhite};
+  border-radius: ${({ theme }) => theme.orbit.borderRadiusNormal};
+  border: solid 1px ${({ theme }) => theme.orbit.paletteCloudNormal};
+  box-shadow: 0 2px 8px 0 rgba(23, 27, 30, 0.1); //TODO Create token boxShadowTile
+  box-sizing: border-box;
+  transition: box-shadow ${({ theme }) => theme.orbit.durationFast} ease-in-out;
+  overflow: hidden;
+
+  &:hover {
+    box-shadow: 0 4px 18px 0 rgba(23, 27, 30, 0.1); //TODO Create token boxShadowTileHover
+  }
+
+  ${StyledIconRight} {
+    transform: ${({ toggle }) => toggle && "rotate(-180deg)"};
+    transition: ${({ theme }) => theme.orbit.durationFast};
+  }
+
+  ${StyledTileExpandableContent} {
+    border-top: ${({ theme, toggle }) =>
+      toggle
+        ? `1px solid ${theme.orbit.paletteCloudNormal}`
+        : `0px solid ${theme.orbit.paletteCloudNormal}`};
+    padding: ${({ theme, toggle }) => toggle && `${theme.orbit.spaceMedium} 0`};
+    margin: 0 ${({ theme }) => theme.orbit.spaceMedium};
+    max-height: ${({ toggle, contentHeight, visible }) =>
+      visible && (toggle ? `${contentHeight}px` : "0")};
+    animation: ${({ theme, toggle, contentHeight, visible }) =>
+      visible &&
+      (toggle
+        ? `${theme.orbit.durationFast} ${toggleDown(contentHeight)} linear`
+        : `${theme.orbit.durationFast} ${toggleUp(contentHeight)} linear`)};
+    transition: max-height ${({ theme }) => theme.orbit.durationFast} linear,
+      padding ${({ theme }) => theme.orbit.durationFast} linear,
+      border-top ${({ theme }) => theme.orbit.durationFast} linear;
+  }
+`;
+
+StyledTileExpandable.defaultProps = {
+  theme: defaultTokens,
+};
+
+class TileExpandable extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    const { shown = false } = this.props;
+    this.state = {
+      visible: !shown,
+      toggle: shown,
+    };
+  }
+
+  state = {
+    toggle: false,
+    visible: false,
+  };
+
+  componentDidMount() {
+    setTimeout(() => this.setHeight(), 250); // Timeout to prevent initial animation from start
+  }
+
+  setHeight() {
+    const { node } = this;
+    if (node instanceof HTMLElement) {
+      const element = Array.from(node.childNodes).find((item, index) => index === 1 && item);
+      if (element) {
+        this.contentHeight = Object.values(element.childNodes).reduce(
+          (acc, child) => child instanceof HTMLElement && acc + child.offsetHeight,
+          0,
+        );
+      }
+    }
+  }
+
+  handleClick = () => {
+    this.setState({ visible: true, toggle: !this.state.toggle });
+  };
+
+  contentHeight = 0;
+  node: ?HTMLElement;
+
+  render() {
+    const { children, shown = false } = this.props;
+    const { contentHeight } = this;
+    return (
+      <StyledTileExpandable
+        contentHeight={contentHeight}
+        visible={this.state.visible}
+        toggle={this.state.toggle}
```

Comment:
Whenever is `toggle` set to `true`, the `Tile` expands (like `expanded`).
On the other hand, `visible` is for a safety check. When you specify `shown` it can be buggy without this state.